src/Compilers/CSharp/Test/Semantic/Semantics/NameLengthTests.cs

```diff
@@ -314,6 +314,8 @@ class C3
                 // (17,24):
                 Diagnostic(ErrorCode.ERR_MetadataNameTooLong, "Short2").WithArguments(s_longSymbolName + 1).WithLocation(17, 24),
                 // (25,24):
+                Diagnostic(ErrorCode.ERR_MetadataNameTooLong, s_longSymbolName + 1).WithArguments(s_longSymbolName + 1).WithLocation(25, 24),
```

Comment:
>Diagnostic(ErrorCode.ERR_MetadataNameTooLong, s_longSymbolName + 1).WithArguments(s_longSymbolName + 1).WithLocation(25, 24), [](start = 16, length = 125)

Can we suppress this duplicate diagnostics, perhaps by checking if the EntryPoint name matches the name of the method?

Reply:
Perhaps even if its length is less than the length of the name of the method.

---
In reply to: [354476273](https://github.com/dotnet/roslyn/pull/40011#discussion_r354476273) [](ancestors = 354476273)